src/charm.py

```diff
@@ -17,6 +17,9 @@
 
 logger = logging.getLogger(__name__)
 
+DEFAULT_LISTEN_PORT = 9119
+DEFAULT_STATS_GROUPS = "server,view,tasks"
```

Comment:
I was thinking about two approach:
 1. using `start` hook
 2. using `if self._stored.empty: self._stored.set_default(...)`

However, both can fail because the charm can be deployed with values other than the default.
`juju deploy my-charm --config name=value`

The only capable solution would be to add the functionality `get_defaults` to `model.CofigData`.

However, `model.ConfigData` uses a backend to [call](https://github.com/canonical/operator/blob/master/ops/model.py#L1521) the juju function `config-get` and this function is not able to print default values.

This could be handy, but the change need to be done at first in juju-tools. I don't know where the repo is or what it's called. Do you?

Reply:
Repo for Operator framework is https://github.com/canonical/operator

Reply:
Yes, but the Operator framework only call the `config-get`, which is function install by juju. (juju-tools)